src/api_server/src/lib.rs

```diff
@@ -30,30 +29,18 @@ use vmm::rpc_interface::{VmmAction, VmmActionError, VmmData};
 use vmm::vmm_config::snapshot::SnapshotType;
 
 use crate::parsed_request::{ParsedRequest, RequestAction};
+use crate::Error::ServerCreation;
 
 /// Shorthand type for a request containing a boxed VmmAction.
 pub type ApiRequest = Box<VmmAction>;
 /// Shorthand type for a response containing a boxed Result.
 pub type ApiResponse = Box<std::result::Result<VmmData, VmmActionError>>;
 
 /// Errors thrown when binding the API server to the socket path.
-#[derive(thiserror::Error)]
```

Comment:
Since this is still an error `Enum`, shouldn't it still derive `thiserror::Error`?

Reply:
The only error this enum is used only to signal that an error during server creation happened, but its value is actually never printed so no need to implement Debug or Display on it.
It is only used as an intermediate to hold the error that triggered the failure of `Http::new()`